text/1185771/higham_przytycka.tex

```diff
@@ -0,0 +1,245 @@
+\\
+We shall denote the distance between two processors $p$ and $q$ with $\delta(p, q)$.
+\\
+We will ocassionally refer to envelopes by their IDs, since they do not change over the algorithm's run.
+\\
+Envelope $b$ is the \it{immediate successor in round i} of envelope $a$ when the first round $i$ envelope encountered after envelope b in round $i$, travelling in the direction of the ring, is envelope $a$.
+Contrary to the paper, we will use lemmas only after proving them, instead of in order of utility.
+\\
+\begin{lemma}\label{p1}
+    A witness for round $k+1$ that promotes an envelope to round $k+2$ was a host for round $k$
+\end{lemma}
+\begin{proof}
+    An envelope with label $c$ and round $k+1$ is promoted to round $k+2$ at its first encounter
+with a processor that has round number $k$ and label, say $d$, less than $c$. The processor that promoted
+envelope $d $to round $k$, $host_k(d)$, has label $d$ and round number $k$ and all other processors with label
+$d$ and round number $k$ must follow $host_k(d)$. Envelope $c$ could not have reached $host_k(d)$ in round $k$
+since otherwise it would have been destroydd by $host_k(d)$, so it encounters $host_k(d)$ in round $k+1$. 
+\end{proof}
+
+\begin{lemma}\label{p2}
+    \[\delta(host_k(b), host_{k+2}(b) \geq F_{k+2} \]
+\end{lemma}
+\begin{proof}
+    Consider $b$’s travel in odd round $k$. If $b$ was promoted to round $k+1$ after travelling
+$F_{k+2}$ links then the observation is immediate. Otherwise, b was promoted by some processor, say
+$host_k(c)$ and $b < c$, and, by the induction hypothesis, $d(host_k(b), host_k(c)) > F_{k+1}$. Since $b$ reaches
+round $k + 2$, in round $k + 1$, $b$ travels from $host_{k+1}(b) (= host_k(c))$ to some witness $w$ for round $k + 1$ with label $d < b$ that promotes $b$ to round $k + 2$. By the inequalities, $c \neq d$, and hence $w$
+must be some processor not in the interval $[host_k(c), host_{k+1}(c))$. Hence $\delta(host_{k+1}(b), host_{k+2}(b))
+\geq \delta(host_k(c), host_{k+1}(c)) \geq F_k$, by the induction hypothesis. The combined distance is therefore at
+least $F_k + F_{k+1} = F_{k+2}$. 
+\end{proof}
+
+\begin{lemma}\label{p3}
+    If envelope $a$ reaches round $i+1$ with $i$ being odd, then $\delta(host_i(a), host_{i+1}(a)) \geq F_{i+1}$
+    If $a$ is destroyed in odd round $i$, then $\delta(host_i(a), destroyer_i(a)) \geq F_i$
+\end{lemma}
+\begin{proof}
+    The proof is by induction on the odd round numbers. The basis, round $1$, holds trivialy because each envelope travels $1 = F_1 = F_2$ link. So suppose that the lemma holds for round $k$ where $k > 1$ and $k$ is odd. Let $a$ and $b$ be labels of two envelopes in round $k + 2$ where envelope $b$ is the immediate successor in round $k + 2$ of envelope $a$. According to the algorithm, envelope $a$ travels $F_{k+4}$ links in round $k + 2$ unless it reaches $host_{k+2}(b)$ before travelling this distance. Therefore, we need to estimate $\delta(host_{k+2}(a), host_{k+2}(b))$.
```

Comment:
trivially - overall, please run a spellchecker on this document, there are a few typos here and there